tree/ntuple/v7/doc/BinaryFormatSpecification.md

```diff
@@ -802,6 +802,27 @@ This section is a comprehensive list of the C++ types with RNTuple I/O support.
 Within the supported type system complex types can be freely composed,
 e.g. `std::vector<MyEvent>` or `std::vector<std::vector<float>>`.
 
+### Type Name Normalization
+
+Type names are stored according to the following normalization rules
+  - All `volatile` qualifiers are removed; `const` qualifiers that do not appear in template arguments are removed.
+  - All whitespaces except between `const` and a template argument are removed.
+  - The `class`, `struct`, and `enum` keywords are removed.
+  - Type names are fully qualified by the namespace in which they are declared.
+  - The integer types `signed char`, `unsigned char`, and `[signed|unsigned](short|int|long[ long])` are replaced by
+    the corresponding (at the time of writing) `std::[u]int(8|16|32|64)_t` standard integer typedef.
+  - Type names used as template arguments are themselves normalized.
+  - For user-defined types, default-initialized template arguments are explicitly spelled out;
+    optional template arguments of stdlib types such as allocators are omitted.
```

Comment:
Note that the I/O normalization we have elected to keep the stdlib optional argument whose value is not the default value.  ie.:
```
TClass::GetClass("vector<int, other_alloc>")->GetName()
(const char *) "vector<int,other_alloc>"
```

Reply:
Ok, I think that should be solved when fixing #16754